Provider/src/FirebirdSql.EntityFrameworkCore.Firebird/Storage/Internal/Mapping/FbByteArrayTypeMapping.cs

```diff
@@ -0,0 +1,106 @@
+/*                 
+ *            FirebirdSql.EntityFrameworkCore.Firebird
+ *              https://www.firebirdsql.org/en/net-provider/ 
+ *     Permission to use, copy, modify, and distribute this software and its
+ *     documentation for any purpose, without fee, and without a written
+ *     agreement is hereby granted, provided that the above copyright notice
+ *     and this paragraph and the following two paragraphs appear in all copies. 
+ * 
+ *     The contents of this file are subject to the Initial
+ *     Developer's Public License Version 1.0 (the "License");
+ *     you may not use this file except in compliance with the
+ *     License. You may obtain a copy of the License at
+ *     http://www.firebirdsql.org/index.php?op=doc&id=idpl
+ *
+ *     Software distributed under the License is distributed on
+ *     an "AS IS" basis, WITHOUT WARRANTY OF ANY KIND, either
+ *     express or implied.  See the License for the specific
+ *     language governing rights and limitations under the License.
+ *
+ *      Credits: Rafael Almeida (ralms@ralms.net)
+ *                              Sergipe-Brazil
+ *                  All Rights Reserved.
+ */
+
+using System;
+using System.Data;
+using System.Data.Common;
+using System.Globalization;
+using System.Text;
+using JetBrains.Annotations;
+using Microsoft.EntityFrameworkCore.Storage;
+
+namespace Microsoft.EntityFrameworkCore.Storage.Internal
+{
+    /// <summary>
+    ///     This API supports the Entity Framework Core infrastructure and is not intended to be used
+    ///     directly from your code. This API may change or be removed in future releases.
+    /// </summary>
+    public class FbByteArrayTypeMapping : ByteArrayTypeMapping
+    {
+        private readonly int _maxSpecificSize;
+
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="FbByteArrayTypeMapping" /> class.
+        /// </summary>
+        /// <param name="storeType"> The name of the database type. </param>
+        /// <param name="dbType"> The <see cref="System.Data.DbType" /> to be used. </param>
+        /// <param name="size"> The size of data the property is configured to store, or null if no size is configured. </param>
+        public FbByteArrayTypeMapping(
+            [NotNull] string storeType,
+            [CanBeNull] DbType? dbType = System.Data.DbType.Binary,
+            int? size = null)
+            : base(storeType, dbType, size)
+        {
+            _maxSpecificSize = CalculateSize(size);
+        }
+
+        private static int CalculateSize(int? size)
+            => size.HasValue && size < 8000 ? size.Value : 8000;
```

Comment:
Are you sure?

Reply:
It will be fixed! this is the default UNICODE SQL Server.